pyscf/lib/dft/grid_common.c

```diff
@@ -255,7 +255,7 @@ static int _orth_components(double *xs_exp, int* bounds, double dx, double radiu
         exp_x0dx *= exp_2dxdx;
     }
 
-    exp_x0dx = exp_dxdx / exp_x0dx_cache;
```

Comment:
I did this to reduce some FLOPs, and it should be fine unless we have a very big exponent. I guess we shouldn't use this function for point charges. @sunqm what do you think?

Reply:
The optimization for flop here is unsafe. It's not just an issue with point charges. exp_x0dx_cache can be zero or infinity for very compact basis functions. The two cases both lead to nan. To avoid this issue, in my early implementation, I explicitly removed the cases where exp_x0x0 is ~0. Additionally, instead of computing these terms as products or divisions, I directly compute both exp(x0x0 + x0dx) and exp(x0x0 - x0dx).
https://github.com/pyscf/pyscf/blob/265b98be4dee8fd59fb3d183d7a99d0b7738ef26/pyscf/lib/dft/numint_uniform_grid.c#L314